File: 0190-reverse-bits/0190-reverse-bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        n=int(n)
        k=31
        num=0
        while(n>0):
            rem=(n&1)
            num+=rem*(2**k)
            k-=1
            n>>=1
        return num
        

File: 0190-reverse-bits/0190-reverse-bits.cpp
int p(int n){
    long long i=1;
    while(n){
        i=i*2;
        n--;
    }
    return i;
}
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int k=31;
        uint32_t s=0;
        while(n){
            int r=n%2;
            s=s+r*(p(k));
            k--;
            n/=2;
        }
        return s;
    }
};